crates/sui-core/src/consensus_validator.rs

```diff
@@ -78,19 +83,24 @@ impl TransactionValidator for SuiTxValidator {
             .map(|tx| tx_from_bytes(tx))
             .collect::<Result<Vec<_>, _>>()?;
 
+        let epoch_store = self.epoch_store.clone();
+
+        let mut owned_tx_certs = Vec::new();
         let mut cert_batch = Vec::new();
         let mut ckpt_messages = Vec::new();
         let mut ckpt_batch = Vec::new();
         for tx in txs.into_iter() {
             match tx.kind {
                 ConsensusTransactionKind::UserTransaction(certificate) => {
-                    cert_batch.push(*certificate);
-
-                    // if !certificate.contains_shared_object() {
-                    //     // new_unchecked safety: we do not use the certs in this list until all
-                    //     // have had their signatures verified.
-                    //     owned_tx_certs.push(VerifiedCertificate::new_unchecked(*certificate));
-                    // }
+                    cert_batch.push(*certificate.clone());
+
+                    if !certificate.contains_shared_object()
+                        && !epoch_store.is_tx_cert_consensus_message_processed(&certificate)?
+                    {
+                        // new_unchecked safety: we do not use the certs in this list until all
+                        // have had their signatures verified.
```

Comment:
can we update the comment here to call out the sigs are verified for `cert_batch` below?